docs/core/extensions/options-validation-generator.md

```diff
@@ -0,0 +1,262 @@
+                (true, false) => "The field {0} must be between {1} exclusive and {2}.",
+                (false, true) => "The field {0} must be between {1} and {2} exclusive.",
+                (true, true) => "The field {0} must be between {1} exclusive and {2} exclusive.",
+            };
+        }
+        private object? ConvertValue(object? value, System.Globalization.CultureInfo formatProvider)
+        {
+            if (value is string stringValue)
+            {
+                value = global::System.Convert.ChangeType(stringValue, OperandType, formatProvider);
+            }
+            else
+            {
+                value = global::System.Convert.ChangeType(value, OperandType, formatProvider);
+            }
+            return value;
+        }
+    }
+}
+
+```
+
+The generated code is optimized for performance and doesn't rely on reflection. The generated code is also AOT-compatible. The generated code is placed in a file named *Validators.g.cs*.
+
+> [!NOTE]
+> Users don't need to take any additional steps to enable the options validation source generator. It is automatically enabled by default when referencing [Microsoft.Extensions.Options](https://www.nuget.org/packages/Microsoft.Extensions.Options) version 8 and above in the project or when building an ASP.NET application.
+
+The only step the app developer needs to take is to add the following to the startup code:
+
+```csharp
+builder.Services
+    .AddSingleton<IValidateOptions<SettingsOptions>, ValidateSettingsOptions>();
+```
+
+> [!NOTE]
+> Calling `ValidateDataAnnotations` is not required when using the options validation source generator.
+
+When the application attempts to access the options object, the generated code for options validation will be executed to validate the options object. The following code snippet illustrates how to access the options object:
+
+```csharp
+    var settingsOptions = app.Services.GetRequiredService<IOptions<SettingsOptions>>().Value;
+```
+
+## Replaced data annotation attributes
+
+Upon close examination of the generated code, you'll observe that the original data annotation attributes, such as <xref:System.ComponentModel.DataAnnotations.RangeAttribute> which were initially applied to the property `SettingsOptions.Scale`, have been substituted with custom attributes like `__SourceGen__RangeAttribute`. This substitution is made because the `RangeAttribute` relies on reflection for validation. In contrast, `__SourceGen__RangeAttribute` is a custom attribute optimized for performance and doesn't depend on reflection, making the code AOT-compatible. The same pattern of attribute replacement will be applied on <xref:System.ComponentModel.DataAnnotations.MaxLengthAttribute>, <xref:System.ComponentModel.DataAnnotations.MinLengthAttribute>, and <xref:System.ComponentModel.DataAnnotations.LengthAttribute> in addition to <xref:System.ComponentModel.DataAnnotations.RangeAttribute>.
```

Comment:
```suggestion
Upon close examination of the generated code, you'll observe that the original data annotation attributes, such as <xref:System.ComponentModel.DataAnnotations.RangeAttribute>, that were initially applied to the property `SettingsOptions.Scale`, have been substituted with custom attributes like `__SourceGen__RangeAttribute`. This substitution is made because the `RangeAttribute` relies on reflection for validation. In contrast, `__SourceGen__RangeAttribute` is a custom attribute optimized for performance and doesn't depend on reflection, making the code AOT-compatible. The same pattern of attribute replacement will be applied on <xref:System.ComponentModel.DataAnnotations.MaxLengthAttribute>, <xref:System.ComponentModel.DataAnnotations.MinLengthAttribute>, and <xref:System.ComponentModel.DataAnnotations.LengthAttribute> in addition to <xref:System.ComponentModel.DataAnnotations.RangeAttribute>.
```